srai/joiners/base.py

```diff
@@ -1,36 +1,24 @@
 """Base class for joiners."""
 
 import abc
-from dataclasses import dataclass
 
 import geopandas as gpd
 
 
-@dataclass
-class JoinerResult:
-    """Joiner result dataclass."""
-
-    regions: gpd.GeoDataFrame
-    features: gpd.GeoDataFrame
-    joined: gpd.GeoDataFrame
-
-
 class BaseJoiner(abc.ABC):
     """Base abstract class for joiners."""
 
     @abc.abstractmethod
-    def join(self, regions: gpd.GeoDataFrame, features: gpd.GeoDataFrame) -> JoinerResult:
+    def join(self, regions: gpd.GeoDataFrame, features: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
         """
         Join features to regions.
 
         Args:
             regions (gpd.GeoDataFrame): regions with which features are joined
             features (gpd.GeoDataFrame): features to be joined
         Returns:
-            JoinerResult object, which contains 3 GeoDataFrames:
-            * original regions
-            * original features
-            * joined features with regions in a MultiIndex with a geometry column
+            GeoDataFrame with an intersection of regions and features, which contains
+            a MultiIndex and a geometry with the intersection
```

Comment:
Are we going to keep the type in Return as well in docstrings? That's how autodocformatter generates the output.
```suggestion
    Returns:
            gpd.GeoDataFrame: GeoDataFrame with an intersection of regions and features, which contains
            a MultiIndex and a geometry with the intersection.
```